pkg/tfbridge/info/info.go

```diff
@@ -666,6 +666,27 @@ type Default struct {
 	EnvVars []string
 }
 
+// ComputeDefaultAutonamingOptionsMode is the mode that controls how the provider handles the proposed name. If not
+// specified, defaults to `Propose`.
+type ComputeDefaultAutonamingOptionsMode int32
+
+const (
+	// ModePropose means the provider may use the proposed name as a suggestion but is free to modify it.
+	ModePropose ComputeDefaultAutonamingOptionsMode = 0
+	// ModeEnforce means the provider must use exactly the proposed name or return an error.
+	ModeEnforce ComputeDefaultAutonamingOptionsMode = 1
+	// ModeDisable means the provider should disable automatic naming and return an error if no explicit name
+	// is provided by user's program.
+	ModeDisable ComputeDefaultAutonamingOptionsMode = 2
```

Comment:
I don't understand what the different between these 2 options is from the provider's perspective.

Can you provide an example where the user sees different behavior with `ModeEnforce` is set vs `ModeDisable`?

Reply:
Enforce will come with a value of proposed name and the provider will use that proposed name to populate the name value if it’s not specified by the user.

Disable will never populate the name if it’s not provided by the user.